### PR TITLE
Initial draft at automatically installing all of the plugins for Mac and Linux

### PR DESCRIPTION
**Pull request by phomer**
Basic bash script.

### CODE REVIEW

File: install_plugins.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+# Init sudo first, if a password is needed
+sudo ls > /dev/null
+
+# TODO: Make sure gvim is installed, put in correct version for MacOS
+
+# Make sure all the directories exist 
+
+# Vim config directory
+if [ ! -d "~/.vim" ]
+then
```
Comment (rigelrozanski) [inline install_plugins.sh]: I'd prefer more compact if statements aka `if [ ! -d "~/.vim" ]; then`

**Comment by rigelrozanski** [comment]
Change readme installation to reflect your updates?

File: install_plugins.sh (hunk 1 above)
Comment (phomer) [inline install_plugins.sh]: Can do. I just didn't initially cause it seems odd and inconsistent to pack only those two statements onto the same line with a separator, but then not do it in general. It's a historically weird convention :-)

**Comment by phomer** [comment]
Should the script be triggered by the makefile? What target?

File: install_plugins.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+#
+# Install all of the plugins for Gvim that are used by this repo.
+# Installation differs between MacOS and Linux
+
+# General setup
+
+set -eu
+
+#OS="MacOS"
```
Comment (rigelrozanski) [inline install_plugins.sh]: We should autodetect the OS, shouldn't have to comment manually 
https://stackoverflow.com/questions/394230/detect-the-os-from-a-bash-script

Comment (rigelrozanski) [inline install_plugins.sh]: Additionally this should detect more OS than just MacOS and Debrian (Ubuntu?) maybe? I guess we can add this in future iterations, but it should be able to distinguish between the two you've built this for automatically

Comment (phomer) [inline install_plugins.sh]: I think the easiest way to do this is (but not necessarily the most correct) is just 'uname'. It returns 'Linux' or 'Darwin'...

Comment (phomer) [inline install_plugins.sh]: Probably won't work for Windows...

Comment (rigelrozanski) [inline install_plugins.sh]: yeah I don't really care about supporting Windows haha

**Pull request merged**